webview/src/experiments/components/table/body/util.ts

```diff
@@ -0,0 +1,32 @@
+import { Row } from '@tanstack/react-table'
+import {
+  Experiment,
+  WORKSPACE_BRANCH
+} from 'dvc/src/experiments/webview/contract'
+
+export const collectBranchWithRows = (
+  rows: Row<Experiment>[]
+): [string | typeof WORKSPACE_BRANCH, Row<Experiment>[]][] => {
+  const branchesWithRows: [
```

Comment:
[F] Usually I would have just used an object here but you can't key an object using `undefined` | `null` so I opted for this slightly wacky collection pattern.